summary/buffer_test.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package summary
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/google/cadvisor/info"
+)
+
+func createSample(i uint64) info.Usage {
+	usage := info.Usage{}
+	usage.PercentComplete = 100
+	usage.Cpu = info.Percentiles{
+		Present: true,
+		Mean:    i * 50,
+		Max:     i * 100,
+		Ninety:  i * 90,
+	}
+	usage.Memory = info.Percentiles{
+		Present: true,
+		Mean:    i * 50 * 1024,
+		Max:     i * 100 * 1024,
+		Ninety:  i * 90 * 1024,
+	}
+	return usage
+}
+
+func expectSize(t *testing.T, b *SamplesBuffer, expectedSize int) {
```

Comment:
nit: not important now, but consider using the assert package we started using in the integration tests. Quite useful and makes the tests pretty readable :)